### PR TITLE
feat(navigation-bar): update component - [FE-1647]

### PR DESCRIPTION
**Pull request by dawidzarzycki**
The commit will update `<NavigationBar />` component to use Styled-Components.
Also commit will replace `primary/secondary` themes with new `light/dark`.
There is no more `transparent` theme.

BREAKING CHANGE:
`as` prop is no more supported as the prop is reserved for Styled-Components.
there is new prop called `navigationType`.
Also `primary` and `secondary` has been replaced with `light` and `dark`.
After update you will need to replace
`<NavigationBar as="primary" />` with `<NavigationBar />`
or
`NavigationBar as="secondary" />` with `<NavigationBar navigationType="dark" />`

### Proposed behaviour
image

### Checklist

- [x] Screenshots are included in the PR
- [x] Carbon implementation and Design System documentation are congruent
- [x] All themes are supported
- [x] Commits follow our style guide
- [x] Unit tests added or updated
- [x] Cypress automation tests added or updated
- [x] Storybook added or updated
- [x] Typescript `d.ts` file added or updated

### CODE REVIEW

File: src/components/navigation-bar/navigation-bar.stories.js [hunk 1]
```diff
@@ -1,47 +1,37 @@
...
-  const component = () => {
-    const children = text('children', '');
-    const as = select(
-      'as',
-      ['primary', 'secondary', 'transparent'],
-      NavigationBar.defaultProps.as
-    );
+export const Basic = () => {
+  const children = text('children', 'Example content');
+  const navigationType = select('navigationType', ['light', 'dark'], NavigationBar.defaultProps.navigationType);
```
Comment (DlgSHi) [inline src/components/navigation-bar/navigation-bar.stories.js:22]: There is no need to pass here those props, we have already rendered those variants in `mdx` and making `chromatic` runs.

File: src/components/navigation-bar/navigation-bar.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import NavigationBar from './navigation-bar.component';
+
+export interface NavigationBarProp {
+  children?: Node;
```
Comment (sjohnson-sage) [inline src/components/navigation-bar/navigation-bar.d.ts:5]: I think `Node` here should be `React.ReactNode`

File: src/components/navigation-bar/navigation-bar.style.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const StyledNavigationBar = styled.div`
+  ${({ navigationType, theme }) => css`
+    min-height: 40px;
+    background-color: ${theme.navigationBar.light.background};
+    border-bottom: 1px solid ${theme.navigationBar.light.borderBottom};
+
+    ${navigationType === 'dark' && css`
+      background-color: ${theme.navigationBar.dark.background};
+      border-bottom:1px solid ${theme.navigationBar.dark.borderBottom};
```
Comment (edleeks87) [inline src/components/navigation-bar/navigation-bar.style.js:12]:
```suggestion
      border-bottom: 1px solid ${theme.navigationBar.dark.borderBottom};
```

File: src/components/navigation-bar/navigation-bar.component.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  );
+};
+
+NavigationBar.propTypes = {
+  children: PropTypes.node,
+  ariaLabel: PropTypes.string,
+  navigationType: PropTypes.oneOf(['light', 'dark'])
+};
+
+NavigationBar.defaultProps = {
```
Comment (sjohnson-sage) [inline src/components/navigation-bar/navigation-bar.component.js:27]: `defaultProps` is being deprecated for functional components - https://github.com/reactjs/rfcs/blob/3416f0de38ee120f6d9dac7f7c9405b77cf37bdc/text/0000-create-element-changes.md#deprecate-defaultprops-on-function-components 
So best to add the default as `navigationType = "light"` up on line 6

Comment (dawidzarzycki) [reply src/components/navigation-bar/navigation-bar.component.js:27]: You are right, I'm happy to follow this one :)

Comment (dawidzarzycki) [reply src/components/navigation-bar/navigation-bar.component.js:27]: I will also use default props in `style.js` file

Comment (dawidzarzycki) [reply src/components/navigation-bar/navigation-bar.component.js:27]: My bad :D It won't work in the `style.js` Will just add this to the main component

**Pull request merged**